x-pack/plugins/lens/public/embeddable/embeddable.tsx

```diff
@@ -468,9 +507,29 @@ export class Embeddable
         this.input.onTableRowClick(event.data as unknown as LensTableRowContextMenuEvent['data']);
       }
     }
+
+    // We allow for edit actions in the Embeddable for display purposes only (e.g. changing the datatable sort order).
+    // No state changes made here with an edit action are persisted.
+    if (isLensEditEvent(event) && this.onEditAction) {
+      if (!this.savedVis) return;
+
+      // have to dance since this.savedVis.state is readonly
+      const newVis = JSON.parse(JSON.stringify(this.savedVis));
```

Comment:
nit: moving the `as Document` from line 519 to this line will give you type safety for line 518

Reply:
Good point. Will do!